server/optimus.go

```diff
@@ -190,11 +204,28 @@ func (s *OptimusServer) setupHTTPProxy() error {
 func (s *OptimusServer) startListening() {
 	// run our server in a goroutine so that it doesn't block to wait for termination requests
 	go func() {
+		// Start listening on grpc server
+		grpcLn := s.mux.MatchWithWriters(
+			cmux.HTTP2MatchHeaderFieldSendSettings("content-type", "application/grpc"))
+		if err := s.grpcServer.Serve(grpcLn); err != nil {
+			s.logger.Info("shutting down grpc server")
+		}
+	}()
+
+	go func() {
+		// Start listening on http server
+		httpL := s.mux.Match(cmux.HTTP1Fast())
```

Comment:
check if it works for http2

Reply:
Used check for grpc and send else to http like before.